compiler/rustc_monomorphize/src/collector.rs

```diff
@@ -775,10 +777,23 @@ impl<'a, 'tcx> MirVisitor<'tcx> for MirNeighborCollector<'a, 'tcx> {
 
         let tcx = self.tcx;
         match terminator.kind {
-            mir::TerminatorKind::Call { ref func, .. } => {
+            mir::TerminatorKind::Call { ref func, ref args, .. } => {
                 let callee_ty = func.ty(self.body, tcx);
                 let callee_ty = self.monomorphize(callee_ty);
-                visit_fn_use(self.tcx, callee_ty, true, source, &mut self.output);
+                match callee_ty.kind() {
+                    ty::FnDef(def_id, _)
+                        if self.tcx.fn_sig(*def_id).skip_binder().abi == RustIntrinsic
+                            && self.tcx.item_name(*def_id) == sym::const_eval_select =>
+                    {
+                        // mark the second arg and the third arg as this function's neighbor.
+                        for func in &args[1..=2] {
```

Comment:
Only the third arg should be necessary, right? The ctfe version shouldn't need to be codegened, does it?